src/providers/twitch/ChatroomChannel.cpp

```diff
@@ -0,0 +1,44 @@
+#include "ChatroomChannel.hpp"
+
+#include <QDebug>
+#include "TwitchApi.hpp"
+#include "common/Common.hpp"
+#include "messages\Emote.hpp"
+#include "providers/bttv/BttvEmotes.hpp"
+#include "providers/bttv/LoadBttvChannelEmote.hpp"
+#include "singletons/Emotes.hpp"
+
+namespace chatterino {
+
+ChatroomChannel::ChatroomChannel(const QString &channelName,
+                                 TwitchBadges &globalTwitchBadges,
+                                 BttvEmotes &globalBttv, FfzEmotes &globalFfz)
+    : TwitchChannel(channelName, globalTwitchBadges, globalBttv, globalFfz)
+{
+    auto list = channelName.split(":");
+    if (list.size() > 2)
```

Comment:
what would be the consequence of a chatroom id such as `chatrooms:123456` being input? the first condition at https://github.com/fourtf/chatterino2/pull/966/files#diff-6f295e4ab67461a536026941daa7aab5R97 would be fulfilled, but then this one wouldn't. would it be smarter to have a freestanding function which ensures the validity of the chatroom syntax before attempting to create a chatroom channel?

Reply:
any particular place where to put the  [isChatroom](https://github.com/fourtf/chatterino2/blob/3c01af980579db1e97c9393446a257a412cf70af/src/providers/twitch/TwitchServer.cpp#L26-L39) function?

Reply:
I think the current positioning is fine :+1: